lib/logstash/inputs/example.rb

```diff
@@ -28,11 +28,17 @@ def register
   end # def register
 
   def run(queue)
+    @thread = Thread.current
```

Comment:
This is not obvious for new plugin developers. I think its worth to add why we need to keep a reference to the current thread inside the `#run` maybe a small explanation even if its related to the `Stud.interval` case.